pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/operator/SortOperatorTest.java

```diff
@@ -0,0 +1,363 @@
+import org.apache.pinot.query.runtime.blocks.TransferableBlock;
+import org.apache.pinot.query.runtime.blocks.TransferableBlockUtils;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static org.apache.pinot.common.utils.DataSchema.ColumnDataType.INT;
+import static org.apache.pinot.common.utils.DataSchema.ColumnDataType.STRING;
+
+
+public class SortOperatorTest {
+
+  private AutoCloseable _mocks;
+
+  @Mock
+  private Operator<TransferableBlock> _input;
+
+  @BeforeMethod
+  public void setUp() {
+    _mocks = MockitoAnnotations.openMocks(this);
+  }
+
+  @AfterMethod
+  public void tearDown()
+      throws Exception {
+    _mocks.close();
+  }
+
+  @Test
+  public void shouldHandleUpstreamErrorBlock() {
+    // Given:
+    List<RexExpression> collation = collation(0);
+    List<Direction> directions = ImmutableList.of(Direction.ASCENDING);
+    DataSchema schema = new DataSchema(new String[]{"sort"}, new DataSchema.ColumnDataType[]{INT});
+    SortOperator op = new SortOperator(_input, collation, directions, 10, 0, schema);
+
+    Mockito.when(_input.nextBlock())
+        .thenReturn(TransferableBlockUtils.getErrorTransferableBlock(new Exception("foo!")));
+
+    // When:
+    TransferableBlock block = op.nextBlock();
+
+    // Then:
+    Assert.assertTrue(block.isErrorBlock(), "expected error block to propagate");
+  }
+
+  @Test
+  public void shouldHandleUpstreamNoOpBlock() {
+    // Given:
+    List<RexExpression> collation = collation(0);
+    List<Direction> directions = ImmutableList.of(Direction.ASCENDING);
+    DataSchema schema = new DataSchema(new String[]{"sort"}, new DataSchema.ColumnDataType[]{INT});
+    SortOperator op = new SortOperator(_input, collation, directions, 10, 0, schema);
+
+    Mockito.when(_input.nextBlock()).thenReturn(TransferableBlockUtils.getNoOpTransferableBlock());
+
+    // When:
+    TransferableBlock block = op.nextBlock();
+
+    // Then:
+    Assert.assertTrue(block.isNoOpBlock(), "expected noop block to propagate");
+  }
+
+  @Test
+  public void shouldCreateEmptyBlockOnUpstreamEOS() {
+    // Given:
+    List<RexExpression> collation = collation(0);
+    List<Direction> directions = ImmutableList.of(Direction.ASCENDING);
+    DataSchema schema = new DataSchema(new String[]{"sort"}, new DataSchema.ColumnDataType[]{INT});
+    SortOperator op = new SortOperator(_input, collation, directions, 10, 0, schema);
+
+    Mockito.when(_input.nextBlock()).thenReturn(TransferableBlockUtils.getEndOfStreamTransferableBlock());
+
+    // When:
+    TransferableBlock block = op.nextBlock();
+
+    // Then:
+    Assert.assertTrue(block.isEndOfStreamBlock(), "expected EOS block to propagate");
+  }
+
+  @Test
+  public void shouldConsumeAndSortInputOneBlockWithTwoRows() {
```

Comment:
I may be missing something. Curious why this test is specialized for 2 rows ?

Reply:
Is there any gotcha we are trying to test for `"one block with 2 rows"` scenario ?

Reply:
yeah - it's essentially testing that it doesn't matter how the rows are distributed amongst the blocks (it's a combine/reduce operator). I could combine it into one test that has `two blocks with two rows each` to cover all that functionality, but this way if one of this or the next test fails we'd know exactly what broke.

let me know if that clarifies